src/service/ToolService.kt

```diff
@@ -42,11 +53,19 @@ class ToolService {
 
             newTool.qualifications = SizedCollection(qualificationsInDb)
 
-            toToolDto(newTool)
+            val toolDto = toToolDto(newTool)
```

Comment:
Maybe use `requireNotNull(deviceInDb)` (see QualificationService)?

Why use let, at all? If deviceInDb was null, the IllegalArgumentException would have been thrown in l.35.